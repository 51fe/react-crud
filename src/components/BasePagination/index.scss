.pagination-container {
  display: flex;
  clear: both;
  background: #fff;

  .el-pagination {
    margin: 20px auto;
  }

  .hidden {
    display: none;
  }
}
