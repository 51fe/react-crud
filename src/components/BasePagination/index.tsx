import './index.scss'
import { Pagination } from 'antd'
import { scrollTo } from '../../utils/scroll-to'

interface BasePaginationProps {
  total: number
  page?: number
  limit?: number
  showSizeChanger?: boolean
  showQuickJumper?: boolean
  background?: boolean
  autoScroll?: boolean
  hidden?: boolean
  pagination?: (page: number, limit: number) => void
}

const BasePagination = ({
  total,
  page = 1,
  limit = 15,
  showSizeChanger,
  showQuickJumper,
  autoScroll = true,
  hidden,
  pagination
}: BasePaginationProps) => {
  const onChange = (current: number, pageSize: number) => {
    pagination?.(current, pageSize)
    if (autoScroll) {
      scrollTo(0, 800)
    }
  }
  return (
    <div className={hidden ? 'pagination-container hidden' : 'pagination-container '}>
      <Pagination
        total={total}
        current={page}
        pageSize={limit}
        pageSizeOptions={[15, 30, 50, 100]}
        showTotal={total => `共 ${total} 条`}
        showSizeChanger={showSizeChanger}
        showQuickJumper={showQuickJumper}
        onChange={onChange}
      />
    </div>
  )
}

export default BasePagination
