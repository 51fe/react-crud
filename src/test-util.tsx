import { ReactElement, ReactNode } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ConfigProvider } from 'antd'
import zhCN from 'antd/lib/locale/zh_CN'
import moment from 'moment'

moment.locale('zh-cn')

const AppProviders = ({ children }: { children: ReactNode }) => {
  return (
    <ConfigProvider locale={zhCN}>
      {children}
    </ConfigProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
): RenderResult => render(
  ui,
  { wrapper: AppProviders, ...options }
)

export * from '@testing-library/react'
export { customRender as render }
export const sleep = (time = 0) =>
  new Promise((resolve) =>
    setTimeout(resolve, time)
  )