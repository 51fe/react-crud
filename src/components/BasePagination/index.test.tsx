import { fireEvent, screen, render } from '../../test-util'
import userEvent from '@testing-library/user-event'
import BasePagination from './index'

test('renders correctly by default', () => {
  const total = 32
  const { container } = render(<BasePagination total={total} />)
  screen.getByText(`共 ${total} 条`)
  expect(container.innerHTML).toMatchSnapshot()
})

test('calls pagination when click the pager number', () => {
  const onPagination = jest.fn()
  const page = 2
  render(<BasePagination total={32} pagination={onPagination} />)
  const numberBtn = screen.getByTitle(page)
  // click page 2 button
  fireEvent.click(numberBtn)
  expect(onPagination).toHaveBeenCalledWith(page, 15)
})

test('calls pagination when click the page size item', async() => {
  const user = userEvent.setup()
  const onPagination = jest.fn()
  const limit = 30
  render(<BasePagination total={32} showSizeChanger pagination={onPagination} />)
  const select = screen.getByRole('combobox', { name: /页码/})
  user.click(select)
  const item = await screen.findByTitle(`${limit} 条/页`)
  // click the second item
  fireEvent.click(item)
  expect(onPagination).toHaveBeenCalledWith(1, limit)
})