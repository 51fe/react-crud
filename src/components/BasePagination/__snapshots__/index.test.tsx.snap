// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly by default 1`] = `"<div class=\\"pagination-container \\"><ul class=\\"ant-pagination\\" unselectable=\\"unselectable\\"><li class=\\"ant-pagination-total-text\\">共 32 条</li><li title=\\"上一页\\" class=\\"ant-pagination-prev ant-pagination-disabled\\" aria-disabled=\\"true\\"><button class=\\"ant-pagination-item-link\\" type=\\"button\\" tabindex=\\"-1\\" disabled=\\"\\"><span role=\\"img\\" aria-label=\\"left\\" class=\\"anticon anticon-left\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"left\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z\\"></path></svg></span></button></li><li title=\\"1\\" class=\\"ant-pagination-item ant-pagination-item-1 ant-pagination-item-active\\" tabindex=\\"0\\"><a rel=\\"nofollow\\">1</a></li><li title=\\"2\\" class=\\"ant-pagination-item ant-pagination-item-2\\" tabindex=\\"0\\"><a rel=\\"nofollow\\">2</a></li><li title=\\"3\\" class=\\"ant-pagination-item ant-pagination-item-3\\" tabindex=\\"0\\"><a rel=\\"nofollow\\">3</a></li><li title=\\"下一页\\" tabindex=\\"0\\" class=\\"ant-pagination-next\\" aria-disabled=\\"false\\"><button class=\\"ant-pagination-item-link\\" type=\\"button\\" tabindex=\\"-1\\"><span role=\\"img\\" aria-label=\\"right\\" class=\\"anticon anticon-right\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"right\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z\\"></path></svg></span></button></li></ul></div>"`;
